### PR TITLE
256 track all blank changes

### PR DESCRIPTION
**Pull request by BrianLMatthews**
Add the `track_blank_changes` option to track changes previously not tracked because both the original and modified values respond with `true` to `blank?`

Changes include adding the option, using the option when looping over `changes`, tests to test the above, documenting the option in `README.md`, and including the change in `CHANGELOG.md`.

### CODE REVIEW

**Comment by BrianLMatthews** [comment]
Something seems to have gone awry when updating CHANGELOG.md the second time, the changes to README.md appear in that commit. All the changes are there, just not as organized as I had it for some reason.

File: CHANGELOG.md
Comment (dblock) [inline CHANGELOG.md:1]: Let's increment the version to 0.9.0 since this is a new feature as part of this PR.

**Review by dblock — CHANGES_REQUESTED** [review]
Overall LGTM! Minor asks below. @johnnyshields should take a look.

File: CHANGELOG.md
Comment (BrianLMatthews) [reply CHANGELOG.md:1]: Will do

**Comment by BrianLMatthews** [comment]
> CI needs fixing, not sure what's up, been a while.

It looks like it's broken with the original code too. The version of `coveralls` it grabs uses `Hash#slice`, which was introduced in ruby 2.5. If I monkey patch that, then it complains about `Fixnum#clamp`, which was introduced in ruby 2.4 or 2.7 (both did something with `clamp`, I didn't read far enough to know which did what). If I monkey patch both:
```
diff --git a/spec/spec_helper.rb b/spec/spec_helper.rb
index 935728f..c3d1094 100644
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -1,3 +1,19 @@
+# rubocop:disable all
+class Hash
+  def slice(*keys)
+    keys.each_with_object({}) { |k, hash| hash[k] = self[k] if key?(k) }
+  end
+end
+
+class Fixnum
+  def clamp(p1, p2)
+    return p1 if self.<=> p1 < 0
+    return p2 if self.<=> p2 > 0
+    self
+  end
+end
+# rubocop:enable all
+
 require 'coveralls'
 Coveralls.wear!
```
then `bundle exec rake` runs fine:
```
$ bundle exec rake
[...]
456 examples, 0 failures, 2 pending
```
So it looks like the gem itself works fine back to at least ruby 2.3.7 (the version I just tried it with), but the specs need 2.5. My preference would be to keep the gem running on older versions and fix the specs.

**Comment by dblock** [comment]
> My preference would be to keep the gem running on older versions and fix the specs.

Whatever gets us to green fastest!

**Comment by BrianLMatthews** [comment]
Is the `coveralls` gem only used for the CI stuff? I uninstalled it, commented out the two lines that reference it in `spec/spec_helper.rb`, and `bundle exec rake` runs just fine and produces the same results. So if `coveralls` isn’t needed that would be the easiest fix. Otherwise the monkey patches in `spec/spec_helper.rb` (maybe with a version check around them) would be next easiest.

Also, i realized I didn’t bump the version in `lib/mongoid/history/version.rb`. Do you want me to do that or do you do that when you release?

**Comment by dblock** [comment]
Coveralls is used to report test coverage in CI. It used to work, so something has broken since and needs to be fixed.

Yes, please increment the version in this PR since this is a new feature. We do the increment after the release so that we don't confuse code that has shipped vs. not.

**Comment by BrianLMatthews** [comment]
Ok, bumped the internal version number, then made a change to `Gemfile` that I think, maybe, fingers crossed, will fix the CI problem. If not I did it in its own commit so it can easily be reverted.

File: CHANGELOG.md
Comment (dblock) [inline CHANGELOG.md:5]: 0.9.0 has not been released, say "0.9.0 (Unreleased)"

**Review by dblock — CHANGES_REQUESTED** [review]
Minor ask for CHANGELOG.

File: CHANGELOG.md
Comment (dblock) [inline CHANGELOG.md:1]: Sorry if I wasn't clear. The next version is 0.9.0, not 0.9.1, please remove the entire 0.9.1 section.

File: CHANGELOG.md
Comment (dblock) [inline CHANGELOG.md:7]: Add a line 

```
* Your contribution here.
```

**Review by dblock — CHANGES_REQUESTED** [review]
Sorry if I wasn't clear about CHANGELOG.

@johnnyshields you're good with this change?

File: CHANGELOG.md
Comment (BrianLMatthews) [reply CHANGELOG.md:7]: Ok, I'm baffled 😂. Here's what I've done to `CHANGELOG.md` so far:
```
$ git diff master CHANGELOG.md
diff --git a/CHANGELOG.md b/CHANGELOG.md
index b55441d..a5ce98c 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,7 +1,11 @@
-### 0.8.6 (Next)
+### 0.9.1 (Next)

 * Your contribution here.

+### 0.9.0 (Unreleased)
+
+* [#257](https://github.com/mongoid/mongoid-history/pull/257): Add track_blank_changes option - [@BrianLMatthews](https://github.com/BrianLMatthews).
+
 ### 0.8.5 (2021/09/18)

 * [#250](https://github.com/mongoid/mongoid-history/pull/250): Migrate to Github actions - [@johnnyshields](https://github.com/johnnyshields).
 ```
That seems like what `CONTRIBUTING.md` is asking for, although I guess it doesn't say to renumber the next release, is that the problem? There is a
```
* Your contribution here.
```
line so I don't understand the request to add one. Maybe just tell me what you want it to look like down to the 0.8.5 line, as obviously I'm not getting something.

Comment (dblock) [reply CHANGELOG.md:7]: This is what it should look like:

```
### 0.9.0 (Next)

* [#257](https://github.com/mongoid/mongoid-history/pull/257): Add track_blank_changes option - [@BrianLMatthews](https://github.com/BrianLMatthews).
* Your contribution here.
```

Comment (BrianLMatthews) [reply CHANGELOG.md:3]: Ok, thanks, updated it to look like that.

**Review by dblock — APPROVED** [review]
Thanks for hanging in here with me @BrianLMatthews. Will merge if I don't hear from @johnnyshields soon :)

**Comment by johnnyshields** [comment]
Its cool! Please go ahead and merge. Thanks!

**Comment by dblock** [comment]
I'm merging this, thank you! 

We have some jruby and 2.7 test failures that look unrelated, @BrianLMatthews maybe you could take a look and add Ruby 3.1/2/3?

**Comment by BrianLMatthews** [comment]
Thanks for merging this! We're working on upgrading our app to Ruby 3.something, when that's done I may be able to take a look. Don't let that stop someone else from looking at it though, don't know when that will be.

Thanks again!

**Comment by niightly** [comment]
Does this fix issue #256?

**Comment by dblock** [comment]
@niightly I believe it does, I'll close it. Open an issue/PR if it doesn't.

**Pull request merged**
